registry/api/v2/urls.go

```diff
@@ -88,6 +88,28 @@ func NewURLBuilderFromRequest(r *http.Request) *URLBuilder {
 	return NewURLBuilder(u)
 }
 
+// NewURLBuilderFromBase uses preconfigured server address information to
+// construct the root url. configScheme and configHost are the URI scheme and
+// hostname, respectively, from the configuration.
+func NewURLBuilderFromBase(r *http.Request, configScheme, configHost string) *URLBuilder {
```

Comment:
I am still wildly confused why we can't just `NewURLBuidlerFromString`? That will allow one to change the full mount of the host path or will just work out of the box. For example, if http.host gets "http://foo.com", it will do the right thing and if it gets "https://foo.com/root/" it will do the thing. No need to require the request as an argument.